library/src/blas3/rocblas_trsm.hpp

```diff
@@ -1435,6 +1486,22 @@ rocblas_status rocblas_trsm_template(rocblas_handle    handle,
     if(batch_count == 0)
         return rocblas_status_success;
 
+    // Temporarily switch to host pointer mode, saving current pointer mode, restored on return
+    auto saved_pointer_mode = handle->push_pointer_mode(rocblas_pointer_mode_host);
+
+    // Get alpha - Check if zero for quick return
+    T alpha_h;
+    if(saved_pointer_mode == rocblas_pointer_mode_host)
+        alpha_h = *alpha;
+    else
+        RETURN_IF_HIP_ERROR(hipMemcpy(&alpha_h, alpha, sizeof(T), hipMemcpyDeviceToHost));
+
+    if(alpha_h == T(0.0))
+    {
+        set_block_unit<T>(handle, m, n, B, ldb, stride_B, batch_count, offset_B);
+        return rocblas_status_success;
+    }
```

Comment:
Added quick return to match Fortran behaviour  - not currently being tests but locally tested